File: Driver Model Shared Control Nominal/UM_11_24_2014/lib/algorithm/sd/cpp/FCN_PAC2002_FY0.cpp
#include <math.h>
#include <stdint.h>
typedef uint16_t char16_t;

// Data
#include "DATA_PAC2002_FY_M.cpp"

// Tire Model
double FCN_PAC2002_FY0(
           double	FZ,
 		   double	ALPHA
		   )
{
    // Pure slip: FY0
    double FY0;
    double ALPHAY, SVY, BY, DY, EY, KY, TMP1, TMP2;
    double SIGN_ALPHAY, SIGN_TMP1;

    ALPHAY  = ALPHA + PH2*FZ + PH1;
    SVY     = PV2*FZ*FZ + PV1*FZ;
    DY      = PD2*FZ*FZ + PD1*FZ;
    SIGN_ALPHAY = ALPHAY/fabs(ALPHAY);
    EY      = (PE2*FZ + PE1)*(1 - PE3*SIGN_ALPHAY);
    KY      = PK1*sin(2*atan(PK2*FZ));
    TMP1    = PD2*PC1*FZ*FZ + PD1*PC1*FZ;
    SIGN_TMP1 = TMP1/fabs(TMP1);
    TMP1    = TMP1 + SIGN_TMP1*0.001;
    BY      = KY/TMP1;
    TMP2    = BY*ALPHAY;
    FY0     = DY*sin(PC1*atan(TMP2 - EY*(TMP2 - atan(TMP2)))) + SVY;

    return FY0;
}

File: Driver Model Shared Control Nominal/UM_11_24_2014/lib/algorithm/opt/OCP_FCN_PAC2002/m/FCN_PAC2002_FY0.m
function FY0 = FCN_PAC2002_FY0(FZ, ALPHA)

[ PC1, PD1, PD2, PE1, PE2, PE3, ...
  PK1, PK2, PH1, PH2, PV1, PV2] = DATA_FY0();

ALPHAY  = ALPHA + PH2*FZ + PH1;
SVY     = PV2*FZ.^2 + PV1*FZ;
DY      = PD2*FZ.^2 + PD1*FZ;
% EY   	= PE2*FZ + PE1 - PE2*PE3*FZ.*sign(ALPHAY) - PE1*PE3*sign(ALPHAY);
EY      = (PE2*FZ + PE1).*(1 - PE3*sign(ALPHAY));
KY      = PK1*sin(2*atan(PK2*FZ));
TMP1    = PD2*PC1*FZ.^2 + PD1*PC1*FZ;
TMP1    = TMP1 + sign(TMP1)*0.001;
BY      = KY./TMP1;
TMP2    = BY.*ALPHAY;
FY0     = DY.*sin(PC1.*atan(TMP2 - EY.*(TMP2 - atan(TMP2)))) + SVY;